test/libsolidity/SolidityNatspecJSON.cpp

```diff
@@ -350,6 +350,51 @@ BOOST_AUTO_TEST_CASE(private_state_variable)
 	checkNatspec(sourceCode, "test", userDoc, true);
 }
 
+BOOST_AUTO_TEST_CASE(event)
+{
+	char const* sourceCode = R"(
+		contract ERC20 {
+			/// @notice This event is emitted when a transfer occurs.
+			/// @param from The source account.
+			/// @param to The destination account.
+			/// @param amount The amount.
+			/// @dev A test case!
+			event Transfer(address indexed from, address indexed to, uint amount);
+		}
+	)";
+
+	char const* devDoc = R"ABCDEF(
+	{
+		"events":
+		{
+			"Transfer(address indexed,address indexed,uint256)":
```

Comment:
@axic Do you have any suggestions?

Reply:
Why do the `indexed` parts need to be stripped? Isn't the point that they are exported as well?

Reply:
As far as I know, they have to be removed when computing the signature of the event.

Reply:
While I'm not happy with how this works, I think the goal was that natspec is used together with the ABI JSON.

In that case a client can find the appropriate entry in the ABI JSON and that has the types, including indexed, listed.

Reply:
Perhaps we could also consider duplicating the ABI JSON under "abi" in the (combined) doc.

Reply:
@axic About your suggestion on duplicating ABI. Does this output look okay?
```JSON
{
  "events": {
    "Transfer(address,address,uint256)": {
      "details": "A test case!",
      "params": {
        "amount": "The amount.",
        "from": "The source account.",
        "to": "The destination account."
      },
      "abi": {
        "anonymous": false,
        "inputs": [
          {
            "indexed": true,
            "internalType": "address",
            "name": "from",
            "type": "address"
          },
          {
            "indexed": true,
            "internalType": "address",
            "name": "to",
            "type": "address"
          },
          {
            "indexed": false,
            "internalType": "uint256",
            "name": "amount",
            "type": "uint256"
          }
        ],
        "name": "Transfer",
        "type": "event"
      }
    }
  }
}

```

Reply:
I thought about just importing the complete ABI JSON as-is under the top level `abi` field, so that clients don't need any new parsing, they can take the `abi` field verbatim.

Reply:
Ok. Then I'll first implement a combined NatSpec format, along with the versioning etc.